lib/core/amount_validator.dart

```diff
@@ -1,32 +1,68 @@
 import 'package:cake_wallet/core/validator.dart';
 import 'package:cake_wallet/generated/i18n.dart';
-import 'package:cw_core/wallet_type.dart';
+import 'package:cw_core/crypto_currency.dart';
 
 class AmountValidator extends TextValidator {
-  AmountValidator({required WalletType type, bool isAutovalidate = false})
+  AmountValidator({required CryptoCurrency currency, bool isAutovalidate = false}) {
+    symbolsAmountValidator =
+        SymbolsAmountValidator(currency: currency, isAutovalidate: isAutovalidate);
+    decimalAmountValidator = DecimalAmountValidator(currency: currency);
+  }
+
+  late final SymbolsAmountValidator symbolsAmountValidator;
+
+  late final DecimalAmountValidator decimalAmountValidator;
+
+  String? call(String? value) => symbolsAmountValidator(value) ?? decimalAmountValidator(value);
+}
+
+class SymbolsAmountValidator extends TextValidator {
+  SymbolsAmountValidator({required CryptoCurrency currency, required bool isAutovalidate})
```

Comment:
we can remove the currency from here since we only check for the symbols, and the `_pattern` function can directly return the 1 RegEx that you are currently using

Reply:
still, the `curreny` parameter can be removed from this class and the `_pattern` can simply return that single RegEx we have
`'^([0-9]+([.\,][0-9]+)?|[.\,][0-9]+)\$'`